pkg/k8s/provider/provider.go

```diff
@@ -0,0 +1,55 @@
+package provider
+
+import (
+	"context"
+
+	"sigs.k8s.io/kind/pkg/cluster/nodes"
+)
+
+type PortMapping struct {
+	HostPort      string
+	ContainerPort string
+}
+
+type KindConfig struct {
+	KindConfigPath string
+}
```

Comment:
better to move the KindConfig definition under `pkg/k8s/providers/kind` for it to be self contained